src/native/corehost/hostpolicy/deps_resolver.cpp

```diff
@@ -735,17 +735,23 @@ void deps_resolver_t::resolve_additional_deps(const arguments_t& args, const dep
 
     for (pal::string_t json_file : m_additional_deps_files)
     {
-        m_additional_deps.push_back(std::unique_ptr<deps_json_t>(
-            new deps_json_t(true, json_file, rid_fallback_graph)));
+        if (rid_fallback_graph != nullptr)
+        {
+            m_additional_deps.push_back(std::unique_ptr<deps_json_t>(
+                new deps_json_t(true, json_file, *rid_fallback_graph)));
+        }
+        else
+        {
+            m_additional_deps.push_back(std::unique_ptr<deps_json_t>(
+                new deps_json_t(true, json_file)));
+        }
     }
 }
 
-void deps_resolver_t::get_app_context_deps_files_range(fx_definition_vector_t::iterator *begin, fx_definition_vector_t::iterator *end) const
+void deps_resolver_t::enum_app_context_deps_files(std::function<void(const pal::string_t&)>& callback)
```

Comment:
If the callback is only used once, then I would pass it in by value instead. The pass by reference logic in this case makes optimizations difficult and doesn't really create a win.